app/data_providers/slate_providers/recommended_reads_slate_provider.py

```diff
@@ -23,6 +24,10 @@ async def rank_corpus_items(
             *args,
             **kwargs,
     ) -> List[CorpusItemModel]:
+        assignment = await self.unleash_provider.get_assignment(POCKET_HOME_V3_FEATURE_FLAG)
+        if assignment is not None and assignment.assigned is True:
+            return items
```

Comment:
Why do you check the assignment here? Isn't that already covered by the code in dispatch.py?

Reply:
We check it here so that we can choose to not thompson sample within the slate

Reply:
Updated logic based on slack convo